tartiflette/engine.py

```diff
@@ -96,79 +96,65 @@ def __init__(
-                of `Engine()`.
-
-                ```python
-                from tartiflette import create_engine
-
-                engine = await create_engine(
-                    sdl,
-                    schema_name = "default",
-                    error_coercer = None,
-                    custom_default_resolver = None,
-                    modules = None,
-                )
-                ```
-
-                More details on the website: https://tartiflette.io/docs/api/engine#create_engine-prepares-and-cooks-your-engine
-                """
-            )
+        self._schema_name = schema_name
+        self._error_coercer = error_coercer
+        self._custom_default_resolver = custom_default_resolver
+        self._modules = modules
+        self._sdl = sdl
 
     async def cook(
         self,
-        sdl: Union[str, List[str]],
+        sdl: Union[str, List[str]] = None,
         error_coercer: Callable[[Exception], dict] = None,
         custom_default_resolver: Optional[Callable] = None,
         modules: Optional[Union[str, List[str]]] = None,
-        schema_name: str = "default",
+        schema_name: str = None,
     ):
         """
         Cook the tartiflette, basicly prepare the engine by binding it to given modules using the schema_name as a key.
         You wont be able to execute a request if the engine wasn't cooked.
 
-        Arguments:
-            sdl {Union[str, List[str]]} -- The SDL to work with.
-
         Keyword Arguments:
+            sdl {Union[str, List[str]]} -- The SDL to work with.
             schema_name {str} -- The name of the SDL (default: {"default"})
             error_coercer {Callable[[Exception, dict], dict]} -- An optional callable in charge of transforming a couple Exception/error into an error dict (default: {default_error_coercer})
             custom_default_resolver {Optional[Callable]} -- An optional callable that will replace the tartiflette default_resolver (Will be called like a resolver for each UNDECORATED field) (default: {None})
             modules {Optional[Union[str, List[str]]]} -- An optional list of string containing the name of the modules you want the engine to import, usually this modules contains your Resolvers, Directives, Scalar or Subscription code (default: {None})
         """
 
         if not modules:
-            modules = []
+            modules = self._modules or []
 
         if isinstance(modules, str):
             modules = [modules]
 
+        schema_name = schema_name or self._schema_name or "default"
+
+        custom_default_resolver = (
+            custom_default_resolver or self._custom_default_resolver
+        )
+        if custom_default_resolver and not iscoroutinefunction(
+            custom_default_resolver
+        ):
+            raise Exception(
+                f"Given custom_default_resolver "
+                f" {custom_default_resolver} "
+                f"is not a coroutine function"
+            )
```

Comment:
I think that

```python
if custom_default_resolver:
    assert iscoroutinefunction(custom_default_resolver)
```

is a simpler implementation. From a user perspective, it would raise an `AssertionError` which would be enough to understand that `custom_default_resolver` must be a coro function. :-)

Reply:
https://docs.python.org/3/reference/simple_stmts.html#the-assert-statement Specify that __debug__ must be true for assert to be evaluated.
As tartiflette doesn't know in which env it will be used, not assert should be used in runtime code.
A lot of python runtime will compile with -O3 arguments thus putting __debug__ to False.

Reply:
Oh, okay, I didn't know about the `-o` argument which disables assertion errors. :+1: